jumpscale/clients/taiga/taiga.py

```diff
@@ -217,3 +224,306 @@ def _get_project_status(project_statuses, status):
+            issues (list): list of issues
+
+        Returns:
+            text(str): the rendered text to be written to markdown file
+
+        """
+        for issue in issues:
+            text += f"- {issue.subject} \n"
+        return text
+
+    def __get_issues_required_data(self, issues, with_description=False):
+        """Get the required data from issues to be used later in export(render)
+
+        Returns:
+            List: List of issues required details to be used in export(render).
+        """
+        all_issues_templates = list()
+        for issue in issues:
+            single_project_template = dict()
+            single_project_template["subject"] = issue.subject
+            single_project_template["created_date"] = issue.created_date
+            single_project_template["due_date"] = issue.due_date
+            single_project_template["owner_name"] = issue.owner_extra_info.get("full_name_display", "unknown")
+            single_project_template["owner_mail"] = issue.owner_extra_info.get("email", "unknown")
+            single_project_template["project"] = issue.project_extra_info.get("name", "unknown")
+            if with_description:
+                single_project_template["description"] = self.get_issue_description(issue.id)
+            all_issues_templates.append(single_project_template)
+        return all_issues_templates
+
+    def export_all_issues_details(self, path="/tmp/issues_details.md", with_description=False):
+        """Export all the issues subjects in a markdown file
+
+        Args:
+            path (str): The path of exported markdown file.
+
+        """
+        text = f"""
+### Issues \n
+"""
+        issues = self.list_all_issues()
+        all_issues_templates = self.__get_issues_required_data(issues, with_description)
+
+        text = self.__render_issues_with_details(
+            text=text, issues=all_issues_templates, with_description=with_description
+        )
+
+        j.sals.fs.write_file(path=path, data=text)
+
+    def __get_single_project_required_data(self, project):
+        single_project_template = dict()
+        single_project_template["name"] = project.name
+        single_project_template["created_date"] = project.created_date
+        single_project_template["owner"] = project.owner.get("full_name_display")
+        single_project_template["issues"] = project.list_issues()
+        return single_project_template
+
+    def __get_project_required_data(self, projects):
+        """Get the required data from the projects
+        Args:
+            projects(list): list of projects
+
+        Returns:
+            all_projects_template (dic): dict of the data required from project object
+
+        """
+        greenlets = [gevent.spawn(self.__get_single_project_required_data, project) for project in projects]
+        gevent.joinall(greenlets)
+        all_projects_template = [
+            greenlet.value for greenlet in greenlets if greenlet.value
```

Comment:
what if the value of the greenlet itself is None? I think better to check on .successful instead of .value